tests/lib/managers/files-test.js

```diff
@@ -2952,4 +2952,144 @@ describe('Files', function() {
 
 	});
 
+	describe('getRepresentationInfo()', function() {
+		var testRepresentation = '[png?dimensions=1024x1024]';
+		var expectedRepresentationParam = { headers: { 'x-rep-hints': testRepresentation }, qs: { fields: 'representations' } };
+		var expectedRepresentationAllParams = { headers: { 'x-rep-hints': testRepresentation, set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png' }, qs: { fields: 'representations' } };
+		var placeholderUrl = '.../{+asset_path}';
+		var representationOptions = {asset_path: '1.png'};
+		var representationAllOptions = {asset_path: '1.png', set_content_disposition_type: 'inline', set_content_disposition_filename: 'New_File_Name.png'};
+
+		it('should make GET request to get file representation when called', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationParam)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationOptions);
+		});
+
+		it('should make GET request to get file representation when called with all optional headers', function() {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [{
+							content: {
+								url_template: placeholderUrl
+							}
+						}]
+					}
+				}
+			};
+			sandbox.mock(boxClientFake).expects('get').withArgs('/files/1234', expectedRepresentationAllParams)
+				.returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationAllOptions);
+		});
+
+		it('should call callback with the representation when a 200 response is returned', function(done) {
+			var response = {
+				statusCode: 200,
+				body: {
+					representations: {
+						entries: [
+							{
+								content: {
+									url_template: placeholderUrl
+								}
+							}
+						]
+					}
+				}
+			};
+			sandbox.stub(boxClientFake, 'get').returns(Promise.resolve(response));
+			files.getRepresentationInfo(FILE_ID, testRepresentation, representationOptions, function(err, representationObject) {
+				assert.ifError(err);
+				assert.strictEqual(representationObject, response.body.representations, 'representation object is returned');
```

Comment:
This tests that the objects are the same (which is good), but it doesn't do any assertion about the asset path replacement.  You should add another assertion to verify that the new `asset_url` property that was added to the object is what you expect.